book/security.rst

```diff
@@ -264,6 +264,18 @@ Great! Now, if you go to ``/admin``, you'll see the HTTP Basic popup:
 .. image:: /images/book/security_http_basic_popup.png
    :align: center
 
+.. caution::
+
+    The ``http_basic`` firewall is only recommended while prototyping applications
+    or when the application is exclusively accesed through secure transports, such
+    as HTTPS. The reason is that browsers include user credentials in each request
+    without applying any hashing mechanism, just a plain base64 encoding.
+
+    Instead, consider using the ``http_digest`` firewall, which is almost identical
+    to ``http_basic`` but where user credentials are encoded and hashed before
+    including them in the request. Read
+    :ref:`HTTP-Digest Authentication reference <reference-security-http-digest>`.
+
```

Comment:
In the Wikipedia, we can find [this list of disadvantages of auth-digest](https://en.wikipedia.org/wiki/Digest_access_authentication#Alternative_authentication_protocols). Is this what you were referring to?

* Many of the security options in RFC 2617 are optional. If quality-of-protection (qop) is not specified by the server, the client will operate in a security-reduced legacy RFC 2069 mode
* Digest access authentication is vulnerable to a man-in-the-middle (MitM) attack. For example, a MitM attacker could tell clients to use basic access authentication or legacy RFC2069 digest access authentication mode. To extend this further, digest access authentication provides no mechanism for clients to verify the server's identity
* Some servers require passwords to be stored using reversible encryption. However, it is possible to instead store the digested value of the username, realm, and password[4]
* It prevents the use of a strong password hash (such as bcrypt) when storing passwords (since either the password, or the digested username, realm and password must be recoverable)